backend/src/plugins/addons/server/services/seed/index.ts

```diff
@@ -0,0 +1,142 @@
+import '@strapi/strapi';
+import { CollectionTypeRepository } from './collection-type-repository';
+import { MediaRepository } from './media-repository';
+import { SingleTypeRepository } from './single-type-repository';
+import { ContentTypeSchema } from './types';
+
+type SeedResult = {
+   contentTypes: Record<
+      string,
+      {
+         count: number;
+      }
+   >;
+   media: {
+      count: number;
+   };
+};
+
+const FALLBACK_STRAPI_ORIGIN = 'https://strapi.ifixit.com';
+
+export default ({ strapi }: { strapi: Strapi.Strapi }) => ({
+   async importContentTypes(
+      strapiOrigin: string = FALLBACK_STRAPI_ORIGIN
+   ): Promise<SeedResult> {
+      const mediaRepo = new MediaRepository({ strapi });
+      const allTypeUIDs = Object.keys(strapi.contentTypes);
+      const apiTypeUIDs = allTypeUIDs.filter((type) =>
+         type.startsWith('api::')
+      );
+      const repos = apiTypeUIDs.map((uid) => {
+         const schema: ContentTypeSchema = strapi.contentType(uid);
+         switch (schema.kind) {
+            case 'collectionType': {
+               return new CollectionTypeRepository({
+                  strapi,
+                  uid,
+               });
+            }
+            case 'singleType': {
+               return new SingleTypeRepository({
+                  strapi,
+                  uid,
+               });
+            }
+         }
+      });
+
+      let shouldSeed = false;
+      for (const repo of repos) {
+         const isRepoSeeded = await repo.isSeeded();
+         if (!isRepoSeeded) {
+            shouldSeed = true;
+            break;
+         }
+      }
+
+      if (repos.length === 0 || !shouldSeed) {
+         strapi.log.info('🌱 Nothing to seed. Skipping..');
+         return { contentTypes: {}, media: { count: 0 } };
+      }
+
+      await mediaRepo.dangerouslyReset();
```

Comment:
Just so I understand correctly: seeding only occurs for previews and local dev? i.e. this code won't be run on strapi.ifixit.com?

Reply:
Good question. Seeding occurs during the bootstrap process. Here's the  the relevant code:

https://github.com/iFixit/react-commerce/blob/improve-previews-seed/backend/src/plugins/addons/server/bootstrap.ts#L4-L5

As you can see the seeding starts only when `SEED_DB` environment variable is set to `true`. So we have to make sure that we're not setting that to `true` in production. The old plugin had the same behaviour, so I assume the production environment is not setting `SEED_DB`, but it's worth double checking.
Additionally, seeding will only occur if at least one content type is empty (which shouldn't be case, although we don't want to rely on this, since in the future we might add new content types.
Another thing to notice is that seeding against production is probably an idempotent operation, because you're effectively inserting the same data (but I won't put this to test because it's not a guarantee).

As a side note, do we have a backup process in place?

Reply:
@danielbeardsley @sterlinghirsh We're discussing this in our standup.

Reply:
@dhmacs You probably can't access this [file](https://github.com/iFixit/server-templates/blob/master/terraform/strapi/prod/main.tf#L64-L77), but we are not setting the `SEED_DB` environment variable in production.

> As a side note, do we have a backup process in place?

@danielbeardsley confirmed that we take snapshots of our Amazon RDS machines daily, so we have backups.

Reply:
I think we all feel comfortable that the content is safe 👍🏻